client/resource_pool.go

```diff
@@ -0,0 +1,42 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	"github.com/google/go-querystring/query"
+	"github.com/maas/gomaasclient/entity"
+)
+
+type ResourcePool struct {
+	ApiClient ApiClient
+}
+
+func (r *ResourcePool) client(id int) ApiClient {
+	return r.ApiClient.GetSubObject(fmt.Sprintf("resourcepools/%v", id))
```

Comment:
According to https://maas.io/docs/api the endpoint to Get/Update/Delete should be in singular form?

> DELETE /MAAS/api/2.0/resourcepool/{id}/
> GET /MAAS/api/2.0/resourcepool/{id}/
> PUT /MAAS/api/2.0/resourcepool/{id}/

```suggestion
	return r.ApiClient.GetSubObject(fmt.Sprintf("resourcepool/%v", id))
```

Reply:
nice catch! Fixed